src/Slick/Utils.hs

```diff
@@ -0,0 +1,36 @@
+module Slick.Utils
+  ( getDirectoryPaths
+  , fmapFmapOptDescr
+  ) where
+
+import           Control.Monad
+import           Data.List                  (intersperse, sortBy)
+import           Data.List                  as L (intersperse, sortBy, (\\))
+import           Data.Ord                   (comparing)
+import qualified Data.Text                  as T
+import           Data.Time
+import           Development.Shake
+import           Development.Shake.FilePath
+import           System.Console.GetOpt
+import           System.Directory
+import           System.Directory.Extra     (listFilesRecursive, removeFile)
+import           System.Environment
+
+------------------------------------------------------------------------------
+-- Helper functions
+
+-- | Given a list of extensions and directories,
+--   find all files that match, and return full paths.
+getDirectoryPaths :: [FilePath]         -- ^ file pattern like *.md
+                  -> [FilePath]         -- ^ directories to look at
+                  -> Action [FilePath]
+getDirectoryPaths extensions dirs =
+  concat <$> mapM getPaths dirs
+    where
+      getPaths :: FilePath -> Action [FilePath]
+      getPaths dir =
+        fmap (dir </>) <$>
+          getDirectoryFiles dir extensions
+
+fmapFmapOptDescr :: (a -> b) -> OptDescr (Either String a) -> OptDescr (Either String b)
```

Comment:
`getDirectoryPaths` seems useful, but this one doesn't seem so general purpose; should this one be made internal?

Reply:
It's related only to the internal use, so yes I believe it should be.

Reply:
exposed in main module